packages/core/src/cli/run-volar-tsc.ts

```diff
@@ -21,16 +44,56 @@ export function run(): void {
     // See discussion here: https://github.com/typed-ember/glint/issues/628
   };
 
-  const main = (): void =>
-    runTsc(require.resolve('typescript/lib/tsc'), options, (ts, options) => {
-      const glintConfig = findConfig(cwd);
+  const createLanguagePlugin = (): LanguagePlugin<URI>[] => {
+    const glintConfig = findConfig(cwd);
+    return glintConfig ? [createEmberLanguagePlugin(glintConfig)] : [];
+  };
 
-      if (glintConfig) {
-        const gtsLanguagePlugin = createEmberLanguagePlugin(glintConfig);
-        return [gtsLanguagePlugin];
-      } else {
-        return [];
+  if (hasSpecificFiles) {
+    // For specific files, create temporary tsconfig that inherits from project config
+    const { tempConfigPath, cleanup } = createTempConfigForFiles(cwd, files);
+    const originalArgv = process.argv;
+    
+    try {
+      // Build TypeScript arguments for single file checking
+      const tscArgs = ['node', 'tsc', '--project', tempConfigPath];
```

Comment:
We can't invoke tsc for type cheking.

That evade's all the transforming glint does.